lib/rules/require-v-if-else-key.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @author Doug Wade <douglas.b.wade@gmail.com>
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const utils = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Helpers
+// ------------------------------------------------------------------------------
+
+// ...
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'require `key` with `v-if/v-else-if/v-else` directives',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/require-v-if-else-key.html'
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      unexpected:
+        'Elements in v-if/v-else-if/v-else expect to have distinct keys if they are of the same type.'
+    }
+  },
+  /** @param {RuleContext} context */
+  create(context) {
+    /**
+     * Gets the key from an element.
+     * @param {VElement} element The element node to get the key from.
+     * @return {String | null}
+     */
+    function getKey(element) {
+      const keyAttribute = utils.getAttribute(element, 'key')
+      if (keyAttribute) {
+        return keyAttribute.value ? keyAttribute.value.value || '' : ''
+      }
+
+      const keyDirective = utils.getDirective(element, 'bind', 'key')
+      if (!keyDirective) {
+        return ''
+      }
+
+      if (
+        !keyDirective.value ||
+        !keyDirective.value.expression ||
+        keyDirective.value.expression.type !== 'Identifier'
+      ) {
+        return null
+      }
+
+      return `v-bind:${keyDirective.value.expression.name}`
+    }
+
+    /**
+     * Check the given element about `v-bind:key` attributes.
+     * @param {VElement} element The element node to check.
+     */
+    function checkKey(element) {
+      const prevSibling = utils.prevSibling(element)
+
+      if (!prevSibling) {
+        return
+      }
+
+      if (prevSibling.name !== element.name) {
+        return
+      }
+
+      const key = getKey(element)
+
+      if (key === null) {
+        return
+      }
+
+      if (!key || key === getKey(prevSibling)) {
```

Comment:
I don't think it is necessary to check the contents of the key. If the user uses the same key, I think the user is intentional.
So, I think we can change it to check if the element has a key.